docs/book/src/entities/registeredservice.md

```diff
@@ -45,49 +46,86 @@ Please note that exclusions have precedence over inclusions.
+
 ## Status
 
 The state of a RegisteredService could be one of the following:
-- Registered
 - Available
 - Claimed
+- Unknown
 - Unreachable
-- Undetermined
-
-> As of now, health-check are still not implemented.
-> The only states that are used are `Available` and `Claimed`.
 
-The RegisteredService state is `Registered` right after creation.
-In most cases this will change immediately after the controller gets notified of existence of new resource, unless controller fails even before it tries to run health-check.
-If the health check passes or is not defined, the state will change to `Available`.
+If a health check is not defined for the service, the state will transition to `Available`.
+Otherwise, the state will be set to `Unknown` while the health check is run for the first time.
+Once the health check completes successfully, the state will transition to `Available`.
+However, if the health check indicates a failure, the state will change to `Unreachable`.
 
-Once a RegisteredService is claimed, its state moves to `Claimed`.
-On the other hand, if the health-check comes back as a failure, the state would change to `Unreachable`.
+In some cases, Primaza cannot actively determine if the health check has succeeded or failed.
+For instance, this can occur if the health check references a non-existent container image, but other cases may also trigger this behavior.
+In those cases, the state of the RegisteredService will transition to `Unknown`, indicating that there may be a problem with the health check and that further investigation may be needed.
 
-In some cases, the health check could fail before it can determine whether the service is healthy or not.
-In those cases, the state of the RegisteredService will go to `Undetermined` indicating that there is an issue with the health-check and not with the service itself.
+Once a RegisteredService is claimed with a [ServiceClaim][ServiceClaim], its state will transition to `Claimed`.
+If the [ServiceClaim][ServiceClaim] is deleted, then the RegisteredService will transition from `Claimed` back to `Available`, making it available for re-use.
 
-There are some situations when the health-check will fail while the RegisteredService is in `Claimed` state.
-In those cases the RegisteredService will move to `Unreachable`.
-If, at a later time, the health-check passes then the controller will check if there is still a claim matching the RegisteredService and move the state back to `Claimed`.
-However, if there is not claim matching the RegisteredService the state will move to `Available`.
+If the health check fails while the RegisteredService is in `Claimed` state, the RegisteredService will transition to `Unreachable`.
+If, at a later time, the health check passes, then the controller will check if there is still a claim matching the RegisteredService and transition to `Claimed`.
+Otherwise, the RegisteredService the state will transition to `Available`.
```

Comment:
@sadlerap  When a RegisteredService transitions from `Claimed` to `Unreachable` what happens to the ServiceBinding?
How can we differentiate a RegisteredService that comes from `Unreachable` and needs to go back to `Claimed` from one that needs to go back to `Available`?

Reply:
My understanding is that we should have the service binding deproject and reproject bindings into the application, so it should try to pick up a different service from the catalog.  I don't know if that's implemented or not, though AIUI a service with a failing healthcheck does get removed from the catalog.

In other words, there's no way for a service to transition from `Unreachable` to `Claimed`.  It can only transition from `Unreachable` to itself (check fails), `Available` (check succeeds), or `Unknown` (check is indeterminate).

Reply:
Healthchecks are tricky to take action on, we need to design this with minimum disruption to the customer. The healthcheck can fail for a few seconds due to connectivity issues that are transitional, they could fail for a few minutes during a short outage or they can fail outright cause the backend service is gone. I think we need to design this with the threshold value in minutes for who long we are willing to wait until the claim controller decides to find another service to match. What I think we can do here no is go from unreachable to available when there is claim for this registered service and go back to claimed if there is a claim for this registered service. This should be possible